plugins/woocommerce/includes/admin/class-wc-admin-settings.php

```diff
@@ -796,20 +799,20 @@ public static function save_fields( $options, $data = null ) {
 
 			// Loop options and get values to save.
 			foreach ( $options as $option ) {
-				if ( ! isset( $option['id'] ) || ! isset( $option['type'] ) || ( isset( $option['is_option'] ) && false === $option['is_option'] ) ) {
+				if ( ! isset( $option['name'] ) || ! isset( $option['type'] ) || ( isset( $option['is_option'] ) && false === $option['is_option'] ) ) {
```

Comment:
At this point, I don't believe `$option['name']` will be set (and so changes to settings will not be saved).